applications/IGAStructuralMechanicsApplication/custom_elements/base_discrete_element.cpp

```diff
@@ -0,0 +1,559 @@
+        }
+
+        KRATOS_CATCH("")
+    };
+
+    /***********************************************************************************/
+    /***********************************************************************************/
+    void BaseDiscreteElement::CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo
+    )
+    {
+        // Calculation flags
+        const bool CalculateStiffnessMatrixFlag = false;
+        const bool CalculateResidualVectorFlag = true;
+        MatrixType temp = Matrix();
+
+        CalculateAll(temp, rRightHandSideVector, rCurrentProcessInfo, CalculateStiffnessMatrixFlag, CalculateResidualVectorFlag);
+    }
+
+    /***********************************************************************************/
+    /***********************************************************************************/
+    void BaseDiscreteElement::CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo
+    )
+    {
+        //calculation flags
+        const bool CalculateStiffnessMatrixFlag = true;
+        const bool CalculateResidualVectorFlag = true;
+
+        CalculateAll(rLeftHandSideMatrix, rRightHandSideVector, rCurrentProcessInfo, CalculateStiffnessMatrixFlag, CalculateResidualVectorFlag);
+    }
+
+    /***********************************************************************************/
+    /***********************************************************************************/
+    void BaseDiscreteElement::CalculateMassMatrix(
+        MatrixType& rMassMatrix,
+        ProcessInfo& rCurrentProcessInfo
+    )
+    {
+        KRATOS_ERROR << "You have called to the CalculateMassMatrix() from the base class BaseDiscreteElement" << std::endl;
+    }
+
+    /***********************************************************************************/
+    /***********************************************************************************/
+    void BaseDiscreteElement::CalculateDampingMatrix(
+        MatrixType& rDampingMatrix,
+        ProcessInfo& rCurrentProcessInfo
+    )
+    {
+        KRATOS_TRY;
+        unsigned int number_of_control_points = GetGeometry().size();
+
+        // Resizing as needed the LHS
+        unsigned int mat_size = number_of_control_points * 3;
+
+        if (rDampingMatrix.size1() != mat_size)
+            rDampingMatrix.resize(mat_size, mat_size, false);
+
+        noalias(rDampingMatrix) = ZeroMatrix(mat_size, mat_size);
+
+        // 1.-Get Damping Coeffitients (RAYLEIGH_ALPHA, RAYLEIGH_BETA)
+        double alpha = 0.0;
+        if (GetProperties().Has(RAYLEIGH_ALPHA))
+            alpha = GetProperties()[RAYLEIGH_ALPHA];
+        else if (rCurrentProcessInfo.Has(RAYLEIGH_ALPHA))
+            alpha = rCurrentProcessInfo[RAYLEIGH_ALPHA];
+
+        double beta = 0.0;
+        if (GetProperties().Has(RAYLEIGH_BETA))
```

Comment:
brackets

Reply:
or use the [ternary operator](https://en.wikipedia.org/wiki/%3F:#C++):
`double beta = GetProperties().Has(RAYLEIGH_BETA) ? GetProperties()[RAYLEIGH_BETA] : 0.0;`